src/control/positionControl/pathPlanning/NumTreesPlanning.cpp

```diff
@@ -1,194 +1,69 @@
             }
         }
-    }
-    if (interface::Output::showDebugNumTreeInfo()) {
-        std::cout << "ROBOT "
-                  << ": reached end of while loop, no path found" << std::endl;
-    }
-    return {};
-}
 
-inline double NumTreesPlanning::remainingStraightLinePathLength(const Vector2 &currentPos, const Vector2 &halfwayPos, const Vector2 &finalPos) {
-    return (abs((halfwayPos - finalPos).length() + (currentPos - halfwayPos).length()));
-}
+        auto collision = collisionDetector.getCollisionBetweenPoints(point.getPosition(), targetPosition);
 
-std::vector<Vector2> NumTreesPlanning::backTrackPath(PathPointer point) {
-    // backtrack the whole path till it hits the root node and return the vector of PathPoints
-    std::vector<Vector2> backTrackedPath = {};
-    while (point) {
-        backTrackedPath.push_back(point->pos);
-        if (point->parent == nullptr) {
+        // no initial collision
+        if (!collision){
+            finalPath = PathPointNode(targetPosition, point);
             break;
         }
-        point.swap(point->parent);
-    }
-
-    // as older points get pushed back, the vector has to be reversed
-    std::reverse(backTrackedPath.begin(), backTrackedPath.end());
-
-    return backTrackedPath;
-}
-
-NumTreesPlanning::PathPointer NumTreesPlanning::computeNewPoint(const Vector2 &targetPosition, const PathPointer &oldPoint, const Vector2 &subTarget) {
-    // Copy parent
-    PathPointer newPoint = std::make_shared<PathPoint>();
-    newPoint->parent = oldPoint;
-    oldPoint->addChild(newPoint);
-    newPoint->t = oldPoint->t + DT;
-    newPoint->currentTarget = subTarget;
-    newPoint->finalTarget = targetPosition;
-    newPoint->pos = oldPoint->pos;
-    newPoint->vel = oldPoint->vel;
-
-    // accelerate towards the target
-    // if the current velocity is away (>90 degrees) from the target, accelerate more towards target
-    Vector2 targetDirection = (subTarget - newPoint->pos);
-    Angle deltaAngle = newPoint->vel.toAngle() - targetDirection.toAngle();
 
-    newPoint->acc = ((targetDirection.normalize() * newPoint->maxVel() - newPoint->vel).normalize() - newPoint->vel.normalize() * fabs(deltaAngle) * M_2_PI);
-
-    Vector2 targetVel = newPoint->vel + newPoint->acc.stretchToLength(DT * std::max(newPoint->maxAcceleration(), newPoint->maxDeceleration()));
-    Angle targetAngle = (targetVel - oldPoint->vel).toAngle();
-
-    // ODE model to get new position/velocity:
-    newPoint->vel = ControlUtils::accelerationLimiter(targetVel, oldPoint->vel, targetAngle, newPoint->maxAcceleration() * DT, newPoint->maxAcceleration() * DT,
-                                                      newPoint->maxDeceleration() * DT, newPoint->maxDeceleration() * DT);
-    newPoint->pos += newPoint->vel * DT;
+        auto branches = branchPath(point, collision.value());
+        std::for_each(branches.begin(), branches.end(), [&pointQueue](const PathPointNode& newPoint){pointQueue.push(newPoint);});
```

Comment:
I think doing this as a lambda makes it less readable than just a plain old foreach